src/tactic/rw_hint/default.lean

```diff
@@ -0,0 +1,46 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+
+import tactic.rewrite_search.discovery
+import tactic.rewrite_all
+
+open tactic.rewrite_all.congr
+open tactic.rewrite_search.discovery
+
+namespace tactic
+
+/-- Generate a list of `rw ...` strings, which represent successful rewrites of the current goal. -/
+-- TODO perhaps run `solve_by_elim` on resulting goals,
```

Comment:
This sounds horrifically slow, no?

Reply:
`solve_by_elim` is scarily fast (Leo did an amazing job with both `apply` and `rw`!) and it would only be running on the successful rewrites, so it's not completely insane. But I haven't tried this.